standards/documenting-infrastructure-owners.md

```diff
@@ -0,0 +1,27 @@
+---
+category: Operating services
+---
+# Documenting owners of infrastructure
+
+All of your infrastructure resources should be tagged with details of its owners, what service it is part of, and what type of environment it is.
+
+Without this, teams responsible for supporting your services (security, finance, and web operations) may not be able to keep your infrastructure secure, paid for by the right people, and appropriately supported. Documenting this on the infrastructure itself ensures that it's as accurate and up-to-date as possible.
+
+You should do this even if you're managing infrastructure in your own account: other teams may end up needing access to that account, or may receive access from another account.
+
+## Tagging your infrastructure
+
+All of our existing cloud hosting providers support tagging ([AWS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html), [Azure](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags), [vCloud](https://blogs.vmware.com/vsphere/2012/03/creating-custom-metadata-using-the-vcloud-api.html)). If your infrastructure is defined in code ([as it should be](https://www.gov.uk/service-manual/technology/manage-your-software-configuration#use-infrastructure-as-code)), you can probably specify your tags in that code.
+
+## Tags you should use
+
+To ensure we can consistently search for, and report on, the tags we use, you should use the following tags. In all values, only use acronyms if you're confident that someone from another part of government would understand them.
+
+- `business-unit`: Should be one of `HQ`, `HMPPS`, `OPG`, `LAA`, `HMCTS`, or `CICA`. If none of these are appropriate, use an appropriate name for the area of the MOJ responsible for the service.
+- `application`: Should be the full name of the application or service (and acronym version, if commonly used), e.g. `Prison Visits Booking`, `Claim for Crown Court Defence (CCCD)`.
+- `component` (optional): Which part of the system this infrastructure is for, e.g. `Staff booking interface`, `API gateway`. If there's a common name for the type of component, use that (e.g. `front-end`, `api`, `message-queue`)
+- `environment`: What type of environment this infrastructure is for. Should be one of `production`, `staging`, `test`, or `development`.
```

Comment:
In the our current implementation `Environment` is used in our ansible code to refer to our AWS account VPC (normally binary choice of `production` or `development` but we also use `user`). We use `Stack` tag to denote prod, staging, qa etc.
The Environment tags are then use to re-lookup some resources, to try to apply idempotently.

I have no objection to the suggest use, but it will mean code changes on a codebase we want to deprecate.

Reply:
I like the idea of having another tag to denote "is production" vs "isn't production", where production means "has actual real data in it".

We often have pre-production or DR environments that are distinct from our production environment, but in the production zone

Reply:
+1 @glenjamin - the list of environment types here is too restrictive, and I'm pretty sure that we only care if an environment is production or not. I'd suggest an optional `environment-name` attribute might be useful to distinguish between them, and a `production` boolean value to indicate if an env contains real data and is in use.

Reply:
👍 I mostly put these in as a straw-man to see what you all said. I like the idea of `is-production` and `environment-name`, so will update to use that.

Reply:
@carinadigital Yeah, this will probably require some re-work of legacy stuff for all of us, but I'd rather we know what's what and spend a bit of extra time on that than not.